x-pack/plugins/lens/public/app_plugin/types.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { History } from 'history';
+import {
+  ApplicationStart,
+  AppMountParameters,
+  ChromeStart,
+  HttpStart,
+  IUiSettingsClient,
+  NotificationsStart,
+  OverlayStart,
+  SavedObjectsStart,
+} from '../../../../../src/core/public';
+import {
+  DataPublicPluginStart,
+  Filter,
+  IndexPattern,
+  Query,
+  SavedQuery,
+} from '../../../../../src/plugins/data/public';
+import { Document } from '../persistence';
+import { LensEmbeddableInput } from '../editor_frame_service/embeddable/embeddable';
+import { NavigationPublicPluginStart } from '../../../../../src/plugins/navigation/public';
+import { LensAttributeService } from '../lens_attribute_service';
+import { IStorageWrapper } from '../../../../../src/plugins/kibana_utils/public';
+import { DashboardFeatureFlagConfig } from '../../../../../src/plugins/dashboard/public';
+import { EmbeddableEditorState } from '../../../../../src/plugins/embeddable/public';
+import { EditorFrameInstance } from '..';
+
+export interface LensAppState {
```

Comment:
Some of these new properties could use a short descriptive comment about what it's expected to do. Especially `indicateNoData`

Reply:
Added a few comments. Should make it a bit more understandable.